quinn/src/lib.rs

```diff
@@ -37,6 +37,58 @@
 //! authority. If this is infeasible--for example, if servers are short-lived or not associated
 //! with a domain name--then as with TLS, self-signed certificates can be used to provide
 //! encryption alone.
+//! 
+//! # Feature flags
+//!
+//! ### Crypto features
+//!
+//! These features control which crypto providers quinn will use. Currently rustls is the only one baked into this crate.
+//! See [quinn-boring](https://github.com/quinn-rs/quinn-boring/) for an example of integrating another crypto provider.
+//!
+//! - `rustls`: Enable rustls as the crypto provider for quinn.
+//!
+//! ### Miscellaneous features
+//!
+//! Miscellaneous optional features.
+//!
+//! - `bloom`: Enabled by default. Enables `BloomTokenLog`, and uses it by default.
+//! - `platform-verifier`: Enabled by default. Provides `ClientConfig::with_platform_verifier()` convenience method.
+//! - `futures-io`: Enables `futures::io::{AsyncRead, AsyncWrite}` support for streams.
+//!
+//! ### Logging features
+//!
+//! This will cause a dependency on the `log` crate and will cause logs to be emitted at
+//! various log levels, for code in quinn or its dependencies.
+//!
+//! - `qlog`: Enabled QUIC logs in the [qlog format](https://quicwg.org/qlog/draft-ietf-quic-qlog-main-schema.html).
+//! - `rustls-log`: Enables logging in the rustls create.
+//! - `lock_tracking`: Records how long locks are held, and warns if they are held >= 1ms.
+//! - `tracing-log`: Configure `tracing` to log events via `log` if no `tracing` subscriber exists.
+//!
+//! ### Crypto features
+//!
+//! The three supported rustls backends are rustls+ring, rustls+aws-lc-rs and rustls+aws-lc-rs-fips.
+//! Note the feature defaults for rustls is to use aws-lc-rs and the default for quinn is to use ring. This means you might 
```

Comment:
Nits: "defaults" -> "default", "one one" -> "on one", "if you don't intend to use multiple rustls backends" -> "if you prefer to stick with a single rustls crypto provider".